packages/suite/src/components/suite/modals/index.tsx

```diff
@@ -116,12 +114,16 @@ const getDeviceContextModal = (props: Props) => {
 
 // Modals requested from `trezor-connect`
 const getDeviceConfirmationModal = (props: Props) => {
-    const { modal, modalActions, goto } = props;
-
+    const { modal, modalActions, goto, router } = props;
     if (modal.context !== MODAL.CONTEXT_DEVICE_CONFIRMATION) return null;
+    const isCoinmarketExchange = router.route?.name === 'wallet-coinmarket-exchange';
 
     switch (modal.windowType) {
         case 'no-backup':
+            // do not show this modal in exchanges - calling getAddress() just for address placeholder
+            // useCoinmarketExchangeForm (line 104)
+            if (isCoinmarketExchange) return null;
```

Comment:
I'd consider creating a new switch case option and use that instead of creating a special case here. But that's just a thing to consider as I don't have clear picture of the architecture yet.

Reply:
@tomexx would you be please so kind and show us what you have in mind in code

Reply:
When I'm looking at it again I see I formulated it incorrectly. My point was if it isn't better to do the decision whether to show or not to show the confirm even before calling this function (= where you call the function) instead of returning `null` here. Or create another switch param for this case and use it where needed. 

But as I wrote before I did not study the architecture of modals yet, so if it doesn't make sense ignore my comment pls.